numpy/core/src/multiarray/number.c

```diff
@@ -397,14 +397,21 @@ is_scalar_with_conversion(PyObject *o2, double* out_exponent)
     PyObject *temp;
     const int optimize_fpexps = 1;
 
-    if (PyInt_Check(o2)) {
-        *out_exponent = (double)PyLong_AsLong(o2);
+    if (PyLong_Check(o2)) {
+        long tmp = PyLong_AsLong(o2);
```

Comment:
This looks like a bug to me - should this not have been long long from the beginning?

Reply:
Ah, it doesn't matter much anyway, it's only checked against tiny integers.

Reply:
Yeah, I considered that too, but came to the same conclusion. I'll take another look just to be sure.

Reply:
Yes, this looks OK.